src/python/grpcio/grpc/_cython/_cygrpc/grpc_gevent.pxd.pxi

```diff
@@ -102,7 +102,7 @@ cdef extern from "src/core/lib/iomgr/timer_custom.h":
 cdef extern from "src/core/lib/iomgr/pollset_custom.h":
   struct grpc_custom_poller_vtable:
     void (*init)()
-    void (*poll)(size_t timeout_ms)
+    grpc_error* (*poll)(size_t timeout_ms) except <grpc_error *>4
```

Comment:
The constant integer `4` here is a bit magical. Can we go ahead and wrap `GRPC_ERROR_CANCELLED` and use it here instead?

Reply:
I'm afraid Cython does not allow using anything other than a literal there. I tried using a variable, a function call, a `const` variable etc, the compiler always complains (in fact, `const` [does not seem to work at all in declarations](https://stackoverflow.com/questions/23873652/how-to-use-const-in-cython)):

```
    cdef grpc_error* run_loop(size_t timeout_ms) except GRPC_ERROR_CANCELLED with gil:
                                                       ^
    ------------------------------------------------------------

    src/python/grpcio/grpc/_cython/_cygrpc/grpc_gevent.pyx.pxi:388:52: Not allowed in a constant expression
```

One alternative that should work however would be to not rely on Cython's `except` syntax to return a value upon failure, but instead set the exception and return the value manually. This would look something like this:

```
try:
    g_event.wait(timeout)
    g_event.clear()
    return grpc_error_none()
except BaseException:
    exc_info = sys.exc_info()
    g_event.clear()
    # Avoid running any Python code after setting the exception
    cpython.PyErr_SetObject(exc_info[0], exc_info[1])
    return grpc_error_cancelled()
```

Reply:
I pushed the above change in the PR branch.